File: 401-500/412/solution.go
//简单逻辑
func fizzBuzz(n int) []string {
    ret := make([]string, n)
    for i:=0; i<n; i++ {
        ok := false
        if (i+1) % 3 == 0 {
            ok = true
            ret[i] = "Fizz"
        }
        if (i+1) % 5 == 0 {
            ok = true
            ret[i] += "Buzz"
        }
        if !ok {
            ret[i] = strconv.Itoa(i+1)
        }
    }
    return ret
}

File: 401-500/412/solution.cpp
//简单逻辑
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ret(n);
        for (int i = 1; i <= n; i++) {
            bool f1 = i % 3 == 0;
            bool f2 = i % 5 == 0;
            if (f1 && f2)
                ret[i - 1] = "FizzBuzz";
            else if (f1)
                ret[i - 1] = "Fizz";
            else if (f2)
                ret[i - 1] = "Buzz";
            else
                ret[i - 1] = to_string(i);
        }
        return ret;
    }
};
